app/controllers/profile/users_controller.rb

```diff
@@ -9,15 +9,26 @@ def edit
 
   def update
     @user = current_user
-    if @user.update(user_params)
+    current_pw = @user.password
+    if @user.update(user_params) && current_pw == @user.password
       flash[:success] = 'Profile successfully updated'
       redirect_to '/profile'
+    elsif @user.update(user_params) && current_pw != @user.password
+      flash[:success] = 'Password updated'
+      redirect_to '/profile'
+    elsif user_params[:password] != user_params[:password_confirmation]
+      flash[:error] = @user.errors.full_messages.to_sentence
```

Comment:
may have a helper method for this... park it for revisions

Reply:
Agree, and I completely re-coded the entire logic just now. I used encapsulation to handle the password updates and user profile data updates in different controllers.